test/typetree.jl

```diff
@@ -0,0 +1,19 @@
+using Enzyme
+using LLVM
+using Test
+
+import Enzyme: typetree, TypeTree, API
+
+const ctx = LLVM.Context()
+const dl = string(LLVM.DataLayout(LLVM.JITTargetMachine()))
+
+tt(T) = string(typetree(T, ctx, dl))
+
+@testset "TypeTree" begin
+    @test tt(Float16) == "{[0]:Float@half}" # TODO(VC) inconsistent
```

Comment:
@wsmoses this should probably be `[-1]` as well?

Reply:
It should be, yes